### ISSUE TITLE
There is a new warning for Firefox extension during linting

### ISSUE THREAD

**Issue by pcdeshmukh** [mozilla/addons-linter#3199]
Hello,
During build we received a new warning `PROP_NAME_NOT_ALLOWED`. What is the intend of the of this warning. Our add-on name is "Acrolinx for Firefox".

```
[INFO] root: Loading suppressions from 'mozlintignore.json'.
 npx: installed 349 in 16.402s
 ========== Warnings ==========
 Code:         PROP_NAME_NOT_ALLOWED
 Message:      The "name" property must not contain the words "mozilla" or "firefox".
 Description:  See http://mzl.la/1STmr48 (MDN Docs) for more information.
 File:         manifest.json
 Line:         <unknown>:<unknown>
 ========== Summary ==========
 Errors:    0
 Warnings:  1
 Notices:   0
```

**Comment by wagnerand** [comment]
Using the words "Firefox" or "Mozilla" in add-on names has not been allowed for a while, this warning makes it more apparent to developers.

**Comment by abhijeetnarvekar** [comment]
@wagnerand "Extension for Firefox" now throws an error for signing the add-on.
` Server response: Add-on names cannot contain the Mozilla or Firefox trademarks. (status: 400)`

The guideline still states "NewGadget Extension for Firefox" is still acceptable.
https://www.mozilla.org/en-US/foundation/trademarks/policy/

Can you clarify?
